pkg/rulesConfig/defaultRules/github/13-prevent-adopt-distributions-on-setup-java.json

```diff
@@ -0,0 +1,67 @@
+{
+	"description": "Prevent use of adopt/adopt-hotspot distributions for actions/setup-java version 2+.",
+	"failureMessage": "adopt/adopt-hotspot distribution is used on the workflow. Please, avoid the use of adopt/adopt-hotspot distributions since they are out of support. Use temurin based distributions instead (such distributions are currently maintained by the core adopt/adopt-hotspot team).",
+	"uniqueId": 13,
+	"enableByDefault": true,
+	"schema": {
+		"type": "object",
+		"additionalProperties": {
+			"type": "object",
+			"properties": {
+				"repositories": {
+					"type": "object",
+					"additionalProperties": {
+						"type": "object",
+						"properties": {
+							"github-actions-workflows": {
+								"type": "object",
+								"additionalProperties": {
+									"properties": {
+										"content": {
+											"properties": {
+												"jobs": {
+													"type": "object",
+													"additionalProperties": {
+														"properties": {
+															"steps": {
+																"type": "array",
+																"items": {
+																	"type": "object",
+																	"if": {
+																		"properties": {
+																			"uses": {
+																				"type": "string",
+																				"pattern": ".*actions\/setup-java@(v2|v3).*"
+																			}
+																		},
+																		"required": [
+																			"uses"
+																		]
+																	},
+																	"then": {
+																		"properties": {
+																			"uses": {
```

Comment:
The format of the if condition looks good. The only thing is that the `distribution` key is given in the `with` section. please see [this example](https://github.com/actions/setup-java#:~:text=v3%0A%2D%20uses%3A%20actions/setup%2Djava%40v3%0A%20%20with%3A-,distribution%3A%20%27temurin%27,-java%2Dversion%3A%20%2717%27%0A%20%20%20%20cache%3A%20%27gradle).

Reply:
Okok! I'll work on the change. Should I access the contents of `with` using `properties` or `additionalProperties`?

Reply:
checkout [this](https://github.com/allero-io/allero/blob/main/pkg/rulesConfig/defaultRules/github/2-ensure-node-version.json) example.